lib/rubocop/cop/style/semicolon.rb

```diff
@@ -78,13 +85,21 @@ def tokens_for_lines
           processed_source.tokens.group_by(&:line)
         end
 
-        def register_semicolon(line, column, after_expression)
+        def register_semicolon(line, column, after_expression, token_before_semicolon = nil)
           range = source_range(processed_source.buffer, line, column)
 
           add_offense(range) do |corrector|
             if after_expression
               corrector.replace(range, "\n")
             else
+              # Prevents becoming one range instance with subsequent line when endless range
+              # without parentheses.
+              # See: https://github.com/rubocop/rubocop/issues/10791
+              if REGEXP_DOTS.include?(token_before_semicolon&.type)
```

Comment:
NOTE: The condition can be described as `token_before_semicolon&.dot_range?` by merging https://github.com/rubocop/rubocop-ast/pull/235.